src/components/twilio-voice-basic-call-control/public/twilio-voice-basic-call-control.js

```diff
@@ -1 +1,107 @@
-// TODO: Implement Web Component
+class TwilioVoiceBasicCallControl extends HTMLElement {
+  #call;
+  #callSid;
+  #conferenceSid;
+
+  constructor() {
+    super();
+    this.attachShadow({ mode: 'open' });
+
+    this.#render();
+    this.#showButtons();
+
+    const twilioVoiceDialer = document.querySelector('twilio-voice-dialer');
+    twilioVoiceDialer.addEventListener('incoming', (e) => {
+      const call = e.detail.call;
+      this.#setCallHandlers(call);
+    });
+    twilioVoiceDialer.addEventListener('outgoing', (e) => {
+      const call = e.detail.call;
+      this.#setCallHandlers(call);
+    });
+
+    this.shadowRoot
+      .querySelector('#hold')
+      .addEventListener('click', () => this.#handleHold());
+    this.shadowRoot
+      .querySelector('#resume')
+      .addEventListener('click', () => this.#handleResume());
+  }
+
+  #handleHold() {
+    this.#setHold(true);
+  }
+
+  #handleResume() {
+    this.#setHold(false);
+  }
+
+  #messageReceivedHandler(message) {
```

Comment:
Nitpick: since all the other handlers are `handleX`, let's call this one `handleCallMessageReceived`?